pkg/build/builder/docker.go

```diff
@@ -69,6 +72,20 @@ func (d *DockerBuilder) Build() error {
 	tag := d.build.Parameters.Output.DockerImageReference
 	defer removeImage(d.dockerClient, tag)
 	if len(d.build.Parameters.Output.DockerImageReference) != 0 {
+		// TODO: This should use ImageRepository ObjectReference in future
```

Comment:
also if it is a thing we wanted to do, then you should have the same TODO for sti.go